### ISSUE TITLE
[Schema Inaccuracy] Create a deployment branch not listing defaults & required.

### ISSUE THREAD

**Issue by kristof-mattei** [github/rest-api-description#3841]
# Schema Inaccuracy

https://docs.github.com/en/rest/deployments/branch-policies?apiVersion=2022-11-28#create-a-deployment-branch-policy

Incorret request & response.

## Expected

* Description of the `type` parameter when absent to default to `branch`.
* Response to list `id`, `node_id`, `name` and `type` as required so that they are not emitted as `type | undefined` in JS.

## Reproduction Steps

Execute

```
gh api \
  --method POST \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  /repos/OWNER/REPO/environments/ENVIRONMENT_NAME/deployment-branch-policies \
   -f "name=release/*"
```

Yields 

```
{
  "id": 20029356,
  "node_id": "MDE2OkdhdGVCcmFuY2hQb2xpY3kyMDAyOTM1Ng==",
  "name": "release/*",
  "type": "branch"
}
```

So omitting `type` creates a `type` = `branch`. Equally there is no way to create a deployment branch policy and not get back and `id`, `node_id`, `name` and a `type`.

**Comment by nickfloyd** [comment]
Please let us know if this work is still needed.  We're trying to follow up on any older issues to make sure they get properly routed.

**Comment by kristof-mattei** [comment]
> Please let us know if this work is still needed. We're trying to follow up on any older issues to make sure they get properly routed.

Yes. The response schema hasn't been fixed yet. Nor has the example response.

**Comment by nickfloyd** [comment]
@kristof-mattei I have routed this to the internal owning API team so that they can have a 👀
